### PR TITLE
[YUNIKORN-2905] Update deployment documentation for make image

### PR DESCRIPTION
**Pull request by chenyao0910**
### What is this PR for?
When I tried to do the same thing in the deployment section, I found that the description for the make image command was outdated.

For example, the IMAGE_TAG variable is no longer exist in the makefile.

Therefore, I've referred to yunikorn-k8shim's readme.md and its makefile, and rewritten the documentation to make it clearer.

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [x] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2905
### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: docs/developer_guide/deployment.md [hunk 1]
```diff
@@ -31,17 +31,22 @@ scheduler and admission controller. It is primarily intended for developers.
 ## Build docker image
 
 Under project root of the `yunikorn-k8shim`, run the command to build an image using the map for the configuration:
+
+Build docker image can be triggered by running following command.
```
Comment (craigcondit) [inline docs/developer_guide/deployment.md:35]: Let's combine with the previous sentence (i.e. ... run the command to build YuniKorn Docker images:

File: docs/developer_guide/deployment.md [hunk 2]
```diff
@@ -31,17 +31,22 @@ scheduler and admission controller. It is primarily intended for developers.
...
 ```
 
-This command will build an image. The image will be tagged with a default version, image tag and your build architecture. 
-
-**Note** the default build uses a hardcoded user and tag. You *must* update the `IMAGE_TAG` variable in the `Makefile` to push to an appropriate repository. 
+**Note** that the default build uses a hardcoded registry and tag. If you want to build docker image with specific arch, version or registry, you can refer the following command.
+```
+make image DOCKER_ARCH=amd64 REGISTRY=apache VERSION=latest
+```
+This command will build an `amd64` binary executable with version `latest` and the docker image tag is `yunikorn/yunikorn:scheduler-amd64-latest`.
```
Comment (craigcondit) [inline docs/developer_guide/deployment.md:45]: Given the example, the tag would be `apache/yunikorn:scheduler-amd64-latest`.

File: docs/developer_guide/deployment.md [hunk 3]
```diff
@@ -31,17 +31,22 @@ scheduler and admission controller. It is primarily intended for developers.
...
-
-**Note** the default build uses a hardcoded user and tag. You *must* update the `IMAGE_TAG` variable in the `Makefile` to push to an appropriate repository. 
+**Note** that the default build uses a hardcoded registry and tag. If you want to build docker image with specific arch, version or registry, you can refer the following command.
+```
+make image DOCKER_ARCH=amd64 REGISTRY=apache VERSION=latest
+```
+This command will build an `amd64` binary executable with version `latest` and the docker image tag is `yunikorn/yunikorn:scheduler-amd64-latest`. 
 
-**Note** the latest yunikorn images in docker hub are not updated anymore due to ASF policy. Hence, you should build both scheduler image and web image locally before deploying them.
+**Note** that the latest yunikorn images in docker hub are not updated anymore due to ASF policy. Hence, you should build both scheduler image and web image locally before deploying them.
```
Comment (craigcondit) [inline docs/developer_guide/deployment.md:47]: Nit: yunikorn -> YuniKorn

**Review by craigcondit — CHANGES_REQUESTED** [review]
A few changes.

**Comment by chenyao0910** [comment]
Thx for your advice, I've pushed an update those wording.

File: docs/developer_guide/deployment.md [hunk 4]
```diff
@@ -30,21 +30,24 @@ scheduler and admission controller. It is primarily intended for developers.
 
 ## Build docker image
 
-Under project root of the `yunikorn-k8shim`, run the command to build an image using the map for the configuration:
+Under project root of the `yunikorn-k8shim`, run the command to build run the command to build YuniKorn Docker images:
```
Comment (craigcondit) [inline docs/developer_guide/deployment.md:33]: "run the command" is listed twice.

File: docs/developer_guide/deployment.md [hunk 5]
```diff
@@ -30,21 +30,24 @@ scheduler and admission controller. It is primarily intended for developers.
...
+
 ```
 make image
 ```
 
-This command will build an image. The image will be tagged with a default version, image tag and your build architecture. 
-
-**Note** the default build uses a hardcoded user and tag. You *must* update the `IMAGE_TAG` variable in the `Makefile` to push to an appropriate repository. 
+**Note** that the default build uses a hardcoded registry and tag. If you want to build docker image with specific arch, version or registry, you can refer the following command.
```
Comment (craigcondit) [inline docs/developer_guide/deployment.md:39]: Grammar: Change "If you want to build docker image with specific arch" to "If you want to build Docker images with a specific arch" and "you can refer the following command." to "you can refer to the following command:"

**Review by craigcondit — CHANGES_REQUESTED** [review]
Minor grammar fixes.

**Comment by chenyao0910** [comment]
Fixed, I need to take some grammar courses lol

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**